src/main/java/org/opensearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -226,17 +224,21 @@ public KNNVectorFieldMapper build(BuilderContext context) {
                         knnMethodContext);
             }
 
-            ModelContext modelContext = this.modelContext.get();
-            if (modelContext != null) {
+            String modelId = this.modelId.get();
+            if (modelId != null) {
+                ModelMetadata modelMetadata = modelDao.getMetadata(modelId);
+
                 return new ModelFieldMapper(
                         name,
-                        new KNNVectorFieldType(buildFullName(context), meta.getValue(), dimension.getValue()),
+                        new KNNVectorFieldType(buildFullName(context), meta.getValue(), modelMetadata.getDimension()),
                         multiFieldsBuilder.build(this, context),
                         copyTo.build(),
                         ignoreMalformed(context),
                         stored.get(),
                         hasDocValues.get(),
-                        modelContext);
+                        modelDao,
+                        modelId,
+                        modelMetadata);
```

Comment:
This can be extracted from modelId and ModelDAO which are passed already. Do we still need to pass?

Reply:
There is a reason, but it is somewhat convoluted. 
1. In 229 we have to get modelMetadata so that we can retrieve the dimension to build the MappedFieldType in 233
2. In 233 we need to build the MappedFieldType here because in 357, KNNVectorFieldMapper constructor takes a MappedFieldType as a construction arg. If we wanted to build MappedFieldType in 600, we would need to call `modelDao.getMetadata(modelId).getDimension()` in KNNVectorFieldType's constructor. Then, we would need to call `modelDao.getMetadata(modelId)` again later in ModelFieldMapper's constructor.

So, tldr, the purpose of this is to prevent duplicate calls to the modelDao.

So, why do we need to pass modelDao then to the ModelFieldMapper's constructor? 
1. In 452, getMergeBuilder passes the modelDao to the builders constructor.
2. In 169, we have ModelDao be a parameter (as opposed to just getting the singleton) to make it easier to test.

I welcome any opinions on any better approaches.